src/main/java/com/uid2/shared/store/parser/ClientParser.java

```diff
@@ -1,25 +1,25 @@
 package com.uid2.shared.store.parser;
 
-import com.uid2.shared.Utils;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.uid2.shared.auth.ClientKey;
-import io.vertx.core.json.JsonArray;
-import io.vertx.core.json.JsonObject;
+import com.uid2.shared.utils.ObjectMapperFactory;
 
 import java.io.IOException;
 import java.io.InputStream;
-import java.util.HashMap;
 import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class ClientParser implements Parser<Map<String, ClientKey>> {
+    private static final ObjectMapper OBJECT_MAPPER = ObjectMapperFactory.build();
+
     @Override
     public ParsingResult<Map<String, ClientKey>> deserialize(InputStream inputStream) throws IOException {
-        JsonArray keysSpec = Utils.toJsonArray(inputStream);
-        Map<String, ClientKey> keyMap = new HashMap<>();
-        for (int i = 0; i < keysSpec.size(); ++i) {
-            JsonObject keySpec = keysSpec.getJsonObject(i);
-            ClientKey clientKey = ClientKey.valueOf(keySpec);
-            keyMap.put(clientKey.getKey(), clientKey);
-        }
-        return new ParsingResult<>(keyMap, keysSpec.size());
+        ClientKey[] clientKeys = OBJECT_MAPPER.readValue(inputStream, ClientKey[].class);
+        Map<String, ClientKey> keyMap = Stream.of(clientKeys).collect(Collectors.toMap(
```

Comment:
Do we need to build this map anymore? Can it be a simple list instead now?

Reply:
I don't think so - its output is used as:
`ParsingResult<T> parsed = parser.deserialize(inputStream);`

And this `T` is part of `ScopedStoreReader<T>`

Reply:
We can change the T to List<ClientKeys>?